From ca991d0c7c844fa2ea86ffddc66089d08c26c246 Mon Sep 17 00:00:00 2001
From: Guillem Jover <guillem@debian.org>
Date: Wed, 24 Nov 2021 23:48:06 +0100
Subject: [PATCH] Dpkg::Vendor::Ubuntu: Update Maintainer field logic to
 include canonical

Some Ubuntu maintainers use addresses with @canonical.com, handle those
in the check.

Closes: #1000557
Based-on-patch-by: William 'jawn-smith' Wilson <william.wilson@canonical.com>
Signed-off-by: Guillem Jover <guillem@debian.org>
---
 scripts/Dpkg/Vendor/Ubuntu.pm | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/scripts/Dpkg/Vendor/Ubuntu.pm b/scripts/Dpkg/Vendor/Ubuntu.pm
index 0352a127d..e222088ff 100644
--- a/scripts/Dpkg/Vendor/Ubuntu.pm
+++ b/scripts/Dpkg/Vendor/Ubuntu.pm
@@ -54,8 +54,8 @@ sub run_hook {
         # https://wiki.ubuntu.com/DebianMaintainerField
         if (defined($fields->{'Version'}) and defined($fields->{'Maintainer'}) and
            $fields->{'Version'} =~ /ubuntu/) {
-           if ($fields->{'Maintainer'} !~ /ubuntu/i) {
-               if (length $ENV{DEBEMAIL} and $ENV{DEBEMAIL} =~ /\@ubuntu\.com/) {
+           if ($fields->{'Maintainer'} !~ /(?:ubuntu|canonical)/i) {
+               if (length $ENV{DEBEMAIL} and $ENV{DEBEMAIL} =~ /\@(?:ubuntu|canonical)\.com/) {
                    error(g_('Version number suggests Ubuntu changes, but Maintainer: does not have Ubuntu address'));
                } else {
                    warning(g_('Version number suggests Ubuntu changes, but Maintainer: does not have Ubuntu address'));
-- 
2.34.0

